src/WebPage.cpp

```diff
@@ -83,13 +83,15 @@ void WebPage::setFrameProperties(QWebFrame *frame, QUrl &requestedUrl, NetworkRe
   if (frame->requestedUrl() == requestedUrl) {
     int statusCode = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
     frame->setProperty("statusCode", statusCode);
-    QVariantMap headers;
-    foreach(QNetworkReply::RawHeaderPair header, reply->rawHeaderPairs())
-      headers[header.first] = QString(header.second);
-    frame->setProperty("headers", headers);
-    frame->setProperty("body", reply->data());
-    QVariant contentMimeType = reply->header(QNetworkRequest::ContentTypeHeader);
-    frame->setProperty("contentType", contentMimeType);
+    if (statusCode != 304) {
```

Comment:
Is the frame object reused (and properties preserved) in the case of a 304? Or, will `headers`, `body` and `contentType` be null/empty for "cached" frames?